core/app/models/spree/stock_item.rb

```diff
@@ -96,12 +96,19 @@ def process_backorders(number)
       end
 
       def conditional_variant_touch
-        # the variant_id changes from nil when a new stock location is added
-        stock_changed = (count_on_hand_changed? && count_on_hand_change.any? { |c| c < 1 }) || variant_id_changed?
-
-        if !Spree::Config.binary_inventory_cache || stock_changed
-          variant.touch
+        # only warn if store is setting binary_inventory_cache (default = false)
+        if Spree::Config.binary_inventory_cache
+          Spree::Config.inventory_cache_threshold = 1
+          ActiveSupport::Deprecation.warn "Spree::Config.binary_inventory_cache=true is DEPRECATED. Instead use Spree::Config.inventory_cache_threshold=1"
         end
+        variant.touch if Spree::Config.inventory_cache_threshold.nil? || count_breaks_threshold?
+      end
+
+      def count_breaks_threshold?
```

Comment:
If this busts cache on `variant_id_changed?`, this method should have a different name.